### PR TITLE
Adds a utils function to parse query params

### PR DESCRIPTION
**Pull request by Ajeyakrishna-k**
Date: 25 December 2025

Developer Name: @Ajeyakrishna-k 

----

## Description: 
- Adds a utility function to parse and convert query params to required format.
- Adds extra validations

Is Under Feature Flag 
- [ ] Yes
- [x] No

Database changes
- [ ] Yes
- [x] No

Breaking changes (If your feature is breaking/missing something please mention pending tickets)
- [ ] Yes
- [x] No

Is Development Tested?

- [x] Yes
- [ ] No

Tested in staging?

- [ ] Yes
- [x] No

### Add relevant Screenshot below ( e.g test coverage etc. )

Screenshot 2023-12-25 at 8 56 09 PM

### CODE REVIEW

File: middlewares/validators/tasks.js [hunk 1]
```diff
@@ -243,6 +245,7 @@ const getUsersValidator = async (req, res, next) => {
           operator: joi.string().valid(Operators.EXCLUDE),
         })
       )
+      .max(20)
```
Comment (ajoykumardas12) [inline middlewares/validators/tasks.js:248]: Does 20 have any significance or arbitrary number?

Comment (Ajeyakrishna-k) [reply middlewares/validators/tasks.js:248]: Approximate number of public holidays we might observe.

**Review by ajoykumardas12 — APPROVED** [review]

File: middlewares/validators/tasks.js [hunk 2]
```diff
@@ -205,7 +205,7 @@ const getUsersValidator = async (req, res, next) => {
   const queryParamsSchema = joi.object().keys({
     cursor: joi.string().optional(),
     q: joi.string().optional(),
-    size: joi.number().integer().min(1).max(2013),
+    size: joi.number().integer().min(1).max(2000),
```
Comment (satyam73) [inline middlewares/validators/tasks.js:208]: This size is used for which thing?

Comment (Ajeyakrishna-k) [reply middlewares/validators/tasks.js:208]: Number of tasks processed.

**Review by satyam73 — APPROVED** [review]
LGTM

**Pull request merged**